odxtools/parameters/tablestructparameter.py

```diff
@@ -14,17 +25,94 @@ def __init__(self, *, table_key_ref, table_key_snref, **kwargs):
             raise OdxError("Either table_key_ref or table_key_snref "
                            "must be defined.")
 
+    def _build_odxlinks(self) -> Dict[OdxLinkId, Any]:
+        return super()._build_odxlinks()
+
+    def _resolve_odxlinks(self, odxlinks: OdxLinkDatabase) -> None:
+        super()._resolve_odxlinks(odxlinks)
+
+        if self.table_key_ref is not None:
+            self._table_key = odxlinks.resolve(self.table_key_ref)
+
+    def _resolve_snrefs(self, diag_layer: "DiagLayer") -> None:
+        super()._resolve_snrefs(diag_layer)
+
+        if self.table_key_snref is not None:
+            raise NotImplementedError("Table keys cannot yet defined using SNREFs"
+                                      " for TableStructParameters.")
+
+    @property
+    def table_key(self) -> TableKeyParameter:
+        return self._table_key
+
     def is_required(self):
-        raise NotImplementedError("TableStructParameter.is_required is not implemented yet.")
+        return True
 
     def is_optional(self):
-        raise NotImplementedError("TableStructParameter.is_optional is not implemented yet.")
+        return False
+
+    def get_coded_value(self, physical_value=None):
+        raise EncodeError("TableStructParameters cannot be converted to "
```

Comment:
Maybe use the same we did for mutex, with the wrapper dict that have one key?

Reply:
I don't really get what you mean here: The issue with table structs is that `physical_value` is depends in which row is selected, but the selected row is not specified in the table struct but the corresponding TABLE-KEY parameter...